app/src/main/java/com/example/caloriecounter/feature/restaurant/ui/categories/CategoriesViewModel.kt

```diff
@@ -0,0 +1,63 @@
+package com.example.caloriecounter.feature.restaurant.ui.categories
+
+import androidx.lifecycle.viewModelScope
+import com.example.caloriecounter.base.BaseViewModel
+import com.example.caloriecounter.base.DataState
+import com.example.domain.handle
+import com.example.domain.usecases.retrofit.GetCategoryListUseCase
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+class CategoriesViewModel @Inject constructor(
+    val getCategoryListUseCase: GetCategoryListUseCase,
+) : BaseViewModel<CategoriesContract.Event, CategoriesContract.State, CategoriesContract.Effect>() {
+
+    init {
+        getCategories()
+    }
```

Comment:
Also question:
Are you planning to duplicate init in every your feature when you want to fetch the data because if so maybe it's better to create some basic function in `BaseViewModel` and then override and call it in your `BaseViewModel`

Reply:
it's overengineering. why do you need to create a `basic function` to do the same as you do in the `init`? plus not all screens require loading from the screen loading so in this case that method would be redundant.

Reply:
That's why I am asking if she is planning to fetch the data in every feature. Okay

Reply:
You are right the only way that is going to help if you then will call a bunch of duplicating functions . We usually call `render` before and after the data is fetched `render` is used in view

Reply:
In my app plans, not all screens require loading from the start.

Reply:
Okay